From db5a6fe731e4c7251074d70ae0cbc39ddc0f4370 Mon Sep 17 00:00:00 2001
From: Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
Date: Fri, 11 Jul 2014 20:59:12 +0200
Subject: [PATCH 2/4] sddm-helper: exit with code 0 otherwise it looks like an
 internal error

When sddm-helper exits with a code other than 0, SDDM::Auth emits the
finished(false) signal which is interpreted as an internal error condition
by the daemon.

However with authentication errors we don't need to exit with an
error code, we just send an error through the socket. It will be
interpreted correctly by the daemon which will display an error in the
greeter.

Issue: #221
---
 src/helper/HelperApp.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/helper/HelperApp.cpp b/src/helper/HelperApp.cpp
index 30d0ef8..5dbd7f1 100644
--- a/src/helper/HelperApp.cpp
+++ b/src/helper/HelperApp.cpp
@@ -113,13 +113,13 @@ void HelperApp::doAuth() {
 
     if (!m_backend->start(m_user)) {
         authenticated(QString(""));
-        exit(AUTH_ERROR);
+        exit(EXIT_SUCCESS);
         return;
     }
 
     if (!m_backend->authenticate()) {
         authenticated(QString(""));
-        exit(AUTH_ERROR);
+        exit(EXIT_SUCCESS);
         return;
     }
 
-- 
2.0.1

